### DISCUSSION TITLE
Display exact versions of interpreters

### DISCUSSION THREAD

**Discussion by gaborbernat** [brettcannon/python-launcher#101] [Ideas]
Sometimes it's handy to know the minor version of the python, I find would be very useful if the ``--version`` would display not just the major version, but also the version data in a third column:

```bash
py --list

┌─────────┬─────────────────────┐
│ Version ┆ Path                │
╞═════════╪═════════════════════╡
│ 3.10    ┆ /usr/bin/python3.10 │
├╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ 3.9     ┆ /usr/bin/python3.9  │
└─────────┴─────────────────────┘
```
For example in the above case the following two values taken from `sys.version`:

```
3.10.0b1 (default, May 17 2021, 07:59:37) [GCC 10.2.0]
3.9.4 (default, Apr 20 2021, 15:51:38) [GCC 10.2.0]
```

**Comment by brettcannon** [comment]
What is the motivation of wanting to have all of this information for all Python interpreters instead of a specific interpreter. E.g. why isn't `py -3.10 --version` or `py -3.10 -c "import sys; print(sys.version)"` not enough? Or is this purely for  convenience?

A key reason I'm asking is as of right now the Launcher doesn't need to run any Python code or do anything with its output. And my experience with the Python extension for VS Code has taught me that plenty of people have custom Python interpreters that spew out a ton of "extra" stuff when launched, which means this unfortunately wouldn't be as simple as simply reading the stdout of executing the `-c` example.

**Reply by gaborbernat** [reply]
Convenience, it would be nice to see all versions installed in one place.

> A key reason I'm asking is, as of right now, the Launcher doesn't need to run any Python code or do anything with its output.

How does python-launcher know the version of the executable `python`?

> And my experience with the Python extension for VS Code has taught me that plenty of people have custom Python interpreters that spew out a ton of "extra" stuff when launched, which means this, unfortunately, wouldn't be as simple as simply reading the stdout of executing the `-c` example.

Guess you'd need to do the same thing virtualenv does, write the information into a temporary JSON file and read that? Granted, this would add some code complexity.

**Comment by brettcannon** [comment]
> How does python-launcher know the version of the executable python?

Parses the file name, e.g. `/usr/bin/python3.8` is `3.8`.

> Guess you'd need to do the same thing virtualenv does, write the information into a temporary JSON file and read that?

Either that or a local socket to ferry the information across.

**Reply by gaborbernat** [reply]
So what about when python3.8 is not available on path, but only unversioned python?

**Reply by brettcannon** [reply]
Unless it's coming from a virtual environment via `$VIRTUAL_ENV` or a `.venv` directory then it is ignored.
